app/services/declarations/create.rb

```diff
@@ -167,5 +168,23 @@ def validate_has_passed?
     def valid_course_identifier_for_participant_outcome?
       CourseGroup.joins(:courses).where(name: %w[leadership specialist]).map(&:courses).flatten.uniq.map(&:identifier).include?(course_identifier)
```

Comment:
@leandroalemao I know this isn't part of this PR, I was thinking this would be better to do this in SQL:
```ruby
CourseGroup.joins(:courses).where(name: %w[leadership specialist], courses: {identifier: course_identifier}).exists?
```
Also maybe with scopes:
```ruby
CourseGroup.joins(:courses).leadership_or_specialist.where(courses: {identifier: course_identifier}).exists?
```